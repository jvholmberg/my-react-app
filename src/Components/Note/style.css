.Note {
  display: block;
  margin-top: .5rem;
  padding: .75rem 1rem;
  width: 100%;
  font-size: 1rem;
  font-weight: 300;
  box-sizing: border-box;
}
.Note.Primary {
  border: none;
  background-color: #f1f1f1;
  border-left: .1rem solid #007bff;
}
.Note.Secondary {
  border: none;
  background-color: #f1f1f1;
  border-left: .1rem solid #6c757d;
}
.Note.Success {
  border: none;
  background-color: #f1f1f1;
  border-left: .1rem solid #28a745;
}
.Note.Danger {
  border: none;
  background-color: #ffe3e5;
  color: #ca3a47;
  border-left: .2rem solid #dc3545;
}
.Note.Warning {
  border: none;
  background-color: #f1f1f1;
  border-left: .1rem solid #ffc107;
}
.Note.Info {
  border: none;
  background-color: #f1f1f1;
  border-left: .1rem solid #17a2b8;
}
.Note.Light {
  background-color: #f1f1f1;
  border: none;
}
.Note.Dark {
  background-color: #bcbcbd;
  border: none;
}
