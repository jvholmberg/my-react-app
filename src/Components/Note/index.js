import React from 'react';
import { CraftClassName } from '../';

import './style.css';

const Note = ({
  primary, secondary, success, danger, warning, info, light, dark,
  className, children, ...props,
}) => {
  const type = { primary, secondary, success, danger, warning, info, light, dark };

  let cn = CraftClassName(type, 'Note', 'Light');
  return (
    <div className={cn} {...props}>
      {children}
    </div>
  );
};
export default Note;
