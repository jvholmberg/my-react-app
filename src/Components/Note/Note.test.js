import React from 'react';

import Note from './';

describe('Note', () => {
  it('Empty matches snapshot', () => {
    expect(shallow(
      <Note />
    )).toMatchSnapshot();
  });
  it('Primary matches snapshot', () => {
    expect(shallow(
      <Note primary />
    )).toMatchSnapshot();
  });
  it('Danger with className matches snapshot', () => {
    expect(shallow(
      <Note className='className' danger />
    )).toMatchSnapshot();
  });
});
