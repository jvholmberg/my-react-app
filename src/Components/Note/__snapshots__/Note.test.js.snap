// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Danger with className matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    className="className"
    danger={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Note Danger",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Note Danger",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Note Empty matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Note Light",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Note Light",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Note Primary matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    primary={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Note Primary",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Note Primary",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
